### DISCUSSION TITLE
Using v-if inside a templae with v-for

### DISCUSSION THREAD

**Discussion by JehhB** [vuejs/petite-vue#206] [Q&A]
I tried using v-if inside a v-for as explained in the Vue.js [documentation](https://vuejs.org/guide/essentials/list.html#v-for-with-v-if).

But it is causing an error in this [part](https://github.com/vuejs/petite-vue/blob/9ff3b98b844033b935143c221176c81244aae32e/src/directives/if.ts#L18) because the element with the v-if directive doesn't have a parent element as it is inside a DocumentFragment rather an HTMLElement.

The v-for directive also uses the same [code](https://github.com/vuejs/petite-vue/blob/9ff3b98b844033b935143c221176c81244aae32e/src/directives/for.ts#L24), so it may also have the same bug. Is there any plan to fix this issue?

**Comment by jimfranke** [comment]
Hi @JehhB, I was running into the same issue and by some investigation I found this bug can be fixed by changing the `parentElement` to `parentNode` within the source of both directives. ParentNode then refers to the `DocumentFragment` instead of being `null`. Best of luck!
